share/functions/__fish_print_VBox_vms.fish

```diff
@@ -0,0 +1,28 @@
+function __fish_print_VBox_vms
+    set -l print_names true
+    set -l print_uuids true
+
+    if set -q argv[1]
+        if test $argv[1] = 'nouuids'
+            set print_uuids false
+        end
+        if test $argv[1] = 'nonames'
+            set print_names false
+        end
+    end
+
+    set -l lines (VBoxManage list vms | string match -r '"(.*)" {(.*)}')
+    while set -q lines[2]
+        if test $print_names = true
+            printf '%s\tVirtual machine\n' $lines[2]
+        end
+
+        if test $print_uuids = true
+            printf '%s\t%s virtual machine\n' $lines[3] $lines[2]
```

Comment:
This uses $lines[3] without checking if it exists. Because of our current (annoying) behavior this may print an error.